ksa_compliance/standard_doctypes/sales_invoice_item.py

```diff
@@ -1,5 +1,11 @@
 import frappe
+import json
 
 
 def calculate_tax_amount(self, method):
-    print(self.as_dict())
+    if self.items:
+        for item in self.items:
+            item_tax_rate = json.loads(item.item_tax_rate)["Expenses Included In Valuation - KD"]
+            item.custom_tax_total = (item.amount * item_tax_rate) / 100
+            item.custom_total_after_tax = item.amount + item.custom_tax_total
+            item.save()
```

Comment:
Here we're saving sales invoice items again, and I'm not happy with this from a performance perspective either. `on_change` is called at the very end, so we end up saving items twice. I'm not sure it would work if the invoice is submitted either. Would saving items raise an exception in that case or be a no-op?